### PR TITLE
Added error controller + separated categories ejs

### PR DESCRIPTION
**Pull request by lyrated**
- added a 404 error
- also copied internal server error function from the book, but if we don't use it we can just ignore or remove it
- separated the category slide into its own ejs file (remember if you use it, you have to pass the variables "categories" from the categories file and "activeCategory" to either NULL or the label name of the selected category)
- added finance as category

### CODE REVIEW

**Comment by simbuzan** [comment]
It looks so good! Links back work well. Code looks good to me. ✔️

**Review by simbuzan — APPROVED** [review]

File: views/error.ejs [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+      <div class="py-5">
+        <a href="/" style="color: #C94744;">
+          <svg xmlns="http://www.w3.org/2000/svg" width="160" height="160" fill="currentColor"
+            class="bi bi-x-circle-fill" viewBox="0 0 16 16">
+            <path
+              d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z" />
+          </svg>
+        </a>
+      </div>
+      <h1 class="h3 font-weight-bold">Something went wrong...</h2>
```
Comment (kimc0de) [inline views/error.ejs:14]: <h1> .... <h2> tag

Comment (kimc0de) [inline views/error.ejs:14]:
```suggestion
      <h1 class="h3 font-weight-bold">Something went wrong...</h1>
```

Comment (lyrated) [reply views/error.ejs]: fixed, thanks<3

Comment (kimc0de) [inline views/error.ejs]: wrong close tag 😆

Comment (lyrated) [reply views/error.ejs]: fixed it thanks :P

**Review by kimc0de — APPROVED** [review]

**Pull request merged**